tests/test_unit/test_apply_for_a_licence/test_views/test_views_start.py

```diff
@@ -1,22 +1,156 @@
 from apply_for_a_licence.choices import WhoDoYouWantTheLicenceToCoverChoices
+from apply_for_a_licence.models import Licence
+from django.test import RequestFactory
 from django.urls import reverse
 
 
+class TestSubmitterReferenceView:
+    def test_post(self, authenticated_al_client):
+        request = RequestFactory().get("/")
+        request.session = authenticated_al_client.session
+        response = authenticated_al_client.post(reverse("submitter_reference"))
+
+        assert response.status_code == 302
+        assert "start" in response.url
+        assert response.url == reverse("start", kwargs={"pk": request.session["licence_id"]})
+
+
 class TestStartView:
     def test_post_myself(self, authenticated_al_client):
+        licence = Licence.objects.create()
+
         response = authenticated_al_client.post(
-            reverse("start"),
+            reverse("start", kwargs={"pk": licence.id}),
             data={"who_do_you_want_the_licence_to_cover": WhoDoYouWantTheLicenceToCoverChoices.myself.value},
         )
 
         assert response.status_code == 302
-        assert response.url == reverse("what_is_your_email")
+        licence_response = Licence.objects.get(pk=licence.id)
+        assert licence_response.who_do_you_want_the_licence_to_cover == "myself"
+        assert response.url == reverse("add_yourself")
 
     def test_post_business(self, authenticated_al_client):
+        licence = Licence.objects.create()
         response = authenticated_al_client.post(
-            reverse("start"),
+            reverse("start", kwargs={"pk": licence.id}),
             data={"who_do_you_want_the_licence_to_cover": WhoDoYouWantTheLicenceToCoverChoices.business.value},
         )
 
         assert response.status_code == 302
+        licence_response = Licence.objects.get(pk=licence.id)
+        assert licence_response.who_do_you_want_the_licence_to_cover == "business"
         assert response.url == reverse("are_you_third_party")
+
+
+class TestThirdPartyView:
+    def test_post_third_party_individual(self, authenticated_al_client):
+        licence = Licence.objects.create(
+            who_do_you_want_the_licence_to_cover=WhoDoYouWantTheLicenceToCoverChoices.individual.value
+        )
+
+        request = RequestFactory().get("/")
+        request.session = authenticated_al_client.session
+        request.session["licence_id"] = licence.id
+        request.session.save()
```

Comment:
I think the below will work, same for all other instances of this:

```suggestion
        session = authenticated_al_client.session
        session["licence_id"] = licence.id
        session.save()
```